plugins/inputs/snmp/snmp.go

```diff
@@ -308,11 +307,10 @@ func (s *Snmp) Gather(acc telegraf.Accumulator) error {
 			return err
 		} else {
 			for _, line := range strings.Split(string(data), "\n") {
-				oidsRegEx := regexp.MustCompile(`([^\t]*)\t*([^\t]*)`)
-				oids := oidsRegEx.FindStringSubmatch(string(line))
-				if oids[2] != "" {
-					oid_name := oids[1]
-					oid := oids[2]
+				oids := strings.Fields(string(line))
```

Comment:
looks like here you need to check for the length of the `oids` slice here, if == 0 then continue the loop I believe?